src/coreclr/jit/emitxarch.cpp

```diff
@@ -11447,11 +11437,25 @@ BYTE* emitter::emitOutputRR(BYTE* dst, instrDesc* id)
     regNumber   reg2 = id->idReg2();
     emitAttr    size = id->idOpSize();
 
-    // Get the 'base' opcode
-    code = insCodeRM(ins);
-    code = AddVexPrefixIfNeeded(ins, code, size);
     if (IsSSEOrAVXInstruction(ins))
     {
+        if (ins == INS_movd)
+        {
+            assert(isFloatReg(reg1) != isFloatReg(reg2));
+            if (isFloatReg(reg1))
+            {
+                code = insCodeRM(ins);
+            }
+            else
+            {
+                code = insCodeMR(ins);
+            }
+        }
+        else
+        {
+            code = insCodeRM(ins);
+        }
+        code = AddVexPrefixIfNeeded(ins, code, size);
```

Comment:
Can we simplify to:
```cpp
assert((ins != INS_movd) || (isFloatReg(reg1) != isFloatReg(reg2)));

if ((ins != INS_movd) || isFloatReg(reg1))
{
    code = insCodeRM(ins);
}
else
{
    code = insCodeMR(ins);
}
```

I'd expect `ins != INS_movd` is the common case for codegen and it helps reduce nesting

Reply:
will do before the merge, thanks for the suggestion.